8323965: modify fix for 8317771 to remove reflection instantiation of the inner class

savoptik opened the pull request:
I replaced reflection with using an accessor
@azuev-java please review

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [ ] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found leading lowercase letter in issue title for `8323965: modify fix for 8317771 to remove reflection instantiation of the inner class`

### Issues
 * [JDK-8323965](https://bugs.openjdk.org/browse/JDK-8323965): modify fix for 8317771 to remove reflection instantiation of the inner class (**Bug** - P4)
 * [JDK-8329667](https://bugs.openjdk.org/browse/JDK-8329667): [macos] Issue with JTree related fix for JDK-8317771 (**Bug** - P3) ⚠️ Issue is already resolved. Consider making this a "backport pull request" by setting the PR title to `Backport <hash>` with the hash of the original commit. See [Backports](https://wiki.openjdk.org/display/SKARA/Backports).


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18867/head:pull/18867` \
`$ git checkout pull/18867`

Update a local copy of the PR: \
`$ git checkout pull/18867` \
`$ git pull https://git.openjdk.org/jdk.git pull/18867/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18867`

View PR using the GUI difftool: \
`$ git pr show -t 18867`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18867.diff">https://git.openjdk.org/jdk/pull/18867.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18867#issuecomment-2066810032)

savoptik commented:
/issue add 8322239

savoptik commented:
/issue remove 8322239

savoptik commented:
/issue add 8329667

liach commented on src/java.desktop/share/classes/sun/swing/SwingAccessor.java:
```diff
@@ -316,6 +325,19 @@ public static void setAccessibleComponentAccessor(final AccessibleComponentAcces
+    public static AccessibleJTreeNodeCreateAccessor getAccessibleJTreeNodeCreateAccessor() {
+        var access = accessibleJTreeNodeCreateAccessor;
+        if (access == null) {
+            ensureClassInitialized(JTree.class);
```
This probably doesn't work as JTree class initialization does not seem to trigger AccessibleJTree class initialization. The other one for AccessibleJTreeNode is dubious too, maybe it just happens that all accesses are correctly made only after the related classes are initialized.

savoptik replied:
Therefore, in the swing accessor, class initialization is forced to be called, which initializes all nested classes. Look at the device of the Neighboring Accessories.

savoptik replied:
This line calls the ```ensureClassInitialized()``` method, which performs a full initialization of the class... It calls: ```MethodHandles.lookup().ensureInitialized()```, which, as stated in the documentation , performs a full initialization.

liach replied:
I know, but initializing a class does not necessarily initialize its nested/inner classes

azuev-java commented:
The problem with this fix is that on the test example attached to the bug any attempt of navigation trough the items of JTree whole voice over is enabled causes java to stop responding. I see in the logs that it does call this exact place thousands of time constantly. So it seems like it makes the problem with java stalling on large size trees to re-appear.

savoptik commented:
@azuev-java 
I'm restoring the context: There was a cycle that recursively collected children, and on new MacOS it worked for a very long time.. JDK-8317771 .
I added a solution for JTree, which works much faster, but there was a reflection, you asked to remove it. Unlike the old algorithm, it now works in seconds, not minutes...
The essence of JDK-8329667 is that the custom tree did not work due to reflection.
I removed the reflection, I did not add any additional acceleration.
As for speeding up the tree, I suggest adding caching, similar to how it is implemented in tables.

azuev-java commented on src/java.desktop/share/classes/javax/swing/JTree.java:
```diff
@@ -4272,6 +4272,10 @@ protected class AccessibleJTree extends AccessibleJComponent
         Accessible leadSelectionAccessible;
 
+        static {
+            SwingAccessor.setAccessibleJTreeNodeCreateAccessor(new AccessibleTreNodeACreateccessor());
```
I think there is a typo here. AccessibleTreNodeACreateccessor does not sound right. Should be AccessibleTreeNodeCreateAccessor.
